lib/state.js

```diff
@@ -242,13 +254,27 @@ StateResolver.prototype.isPathInNodeModulesDirectory_ = function(path) {
 };
 
 
-StateResolver.prototype.resolveFrame_ = function(frame) {
-  var args = this.resolveArgumentList_(frame);
+StateResolver.prototype.resolveFrame_ = function(frame, resolveVars) {
+  var args;
+  var locals;
+  if (resolveVars) {
+    args = this.resolveArgumentList_(frame);
+    locals = this.resolveLocalsList_(frame, args);
+  } else {
+    args = [{
+      name: 'arguments_not_available',
+      varTableIndex: ARG_LOCAL_LIMIT_MESSAGE_INDEX
+    }];
+    locals = [{
+      name: 'locals_not_available',
+      varTableIndex: ARG_LOCAL_LIMIT_MESSAGE_INDEX
+    }];
+  }
```

Comment:
If it doesn't get too unwieldy, I would still prefer a `args = resolveVars ? ... : ...` style initialization.